### ISSUE TITLE
Problem: Support for multiple dublin core file metadata

### ISSUE THREAD

**Issue by camlyall** [artefactual-labs/a3m#345]
Currently only a single json dictionary is supported in the dublin core metadata, dc.json, allowing dc metadata for only a single file in the submission.

Support should be provided for a json array containing multiple file dc metadata.

[dc_metadata_examples.zip](https://github.com/artefactual-labs/a3m/files/11633954/dc_metadata_examples.zip)

Archivematica documentation contains an example of item level descriptive metadata for multiple individual files
https://www.archivematica.org/en/docs/archivematica-1.13/user-manual/transfer/import-metadata/#import-metadata

**Comment by camlyall** [comment]
After some further testing I have discovered additional inconsistencies between a3m and archivematica DC metadata submission.  

[dc_single_json_example.zip](https://github.com/artefactual-labs/a3m/files/11788608/dc_single_json_example.zip)

The above folder contains the submission (beep-boop-uuid) and the resulting AIP (example-uuid).
This demonstrates the following:
- A3M doesn't expect the 'filename' field in dc submission
- A3M doesn't expect the 'dc.' prefix on metadata fields

**Comment by sallain** [comment]
Thanks for this sample, @camlyall! I'll take a look once I have an a3m instance running. I'm curious about your use of `dc.json` as the metadata file name - per the [Archivematica docs](https://www.archivematica.org/en/docs/archivematica-1.14/user-manual/transfer/import-metadata/#metadata-json), the file should be named metadata.json, regardless of the contents. The same should be true in a3m. I wonder if that's a contributing factor here.

Archivematica doesn't require the `dc.` prefix on metadata fields - if you use it, Archivematica will recognize the metadata as Dublin Core and apply `MDTYPE="DC"` in the METS, but if you don't use the prefix it will apply `MDTYPE="OTHER"` in the METS. This is the intended functionality (see [Adding Non-Dublin Core Descriptive Metadata](https://www.archivematica.org/en/docs/archivematica-1.14/user-manual/transfer/import-metadata/#non-dc-metadata).)

**Comment by sallain** [comment]
Running the standard Archivematica metadata.json sample ([DemoTransferJSON](https://github.com/artefactual/archivematica-sampledata/tree/master/SampleTransfers/DemoTransferJSON)) proves that a3m can in fact handle JSON metadata for multiple digital objects. The resulting METS includes the appropriate dmdSecs for each digital object, with the relevant metadata. I have also tested the inclusion of non-DC metadata and it is also written to the METS as expected.

I believe that the problems that @camlyall reported are due to some inconsistencies in both the structure and the metadata in the sample provided. The structure of the original sample is as follows:

```
beep-boop-original
├── data
│   └── beep-boop.png
└── metadata
    └── dc.json
```

I changed the `/data` directory to `/objects` and renamed `dc.json` to `metadata.json`, per the Archivematica documentation:

```
beep-boop-restructured/
├── metadata
│   └── metadata.json
└── objects
    └── beep-boop.png
```

The original metadata file, dc.json, was structured like this:

```
{
    "creator": "creator",
    "title": "title",
    "dc.subject": "subject",
    "unsupported": "unsupported"
}
```

Per the Archivematica documentation, I added a filename element, and also structured the JSON as an array, even though in this instance it is only describing one object:

```
[
    {
        "filename": "objects/beep-boop.png",
        "creator": "creator",
        "title": "title",
        "dc.subject": "subject",
        "unsupported": "unsupported"
    }
]
```

This results in a METS file containing all of the descriptive metadata, in the appropriate dmdSec. The three non-DC fields (creator, title, and unsupported) are tagged as MDTYPE="OTHER" while the Dublin Core field (dc.subject) is MDTYPE="DC".

```
<mets:dmdSec ID="dmdSec_3">
    <mets:mdWrap MDTYPE="OTHER" OTHERMDTYPE="CUSTOM">
      <mets:xmlData>
        <creator>creator</creator>
        <title>title</title>
        <unsupported>unsupported</unsupported>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
  <mets:dmdSec ID="dmdSec_4">
    <mets:mdWrap MDTYPE="DC">
      <mets:xmlData>
        <dcterms:dublincore xmlns:dcterms="http://purl.org/dc/terms/" xmlns:dc="http://purl.org/dc/elements/1.1/" xsi:schemaLocation="http://purl.org/dc/terms/ https://dublincore.org/schemas/xmls/qdc/2008/02/11/dcterms.xsd">
          <dc:subject>subject</dc:subject>
        </dcterms:dublincore>
      </mets:xmlData>
    </mets:mdWrap>
  </mets:dmdSec>
```

In summary, I think the structure of the test transfer resulted in some inconsistencies that can be considered edge-cases. Restructuring the transfer to conform to Archivematica standards produces expected results.
